profiler/matchers.rst

```diff
@@ -6,158 +6,5 @@ How to Use Matchers to Enable the Profiler Conditionally
-``true`` otherwise. Therefore, the custom matcher must return ``false`` to
-disable the profiler and ``true`` to enable it.
-
-Suppose that the profiler must be enabled whenever a user with a
-``ROLE_SUPER_ADMIN`` is logged in. This is the only code needed for that custom
-matcher::
-
-    // src/Profiler/SuperAdminMatcher.php
-    namespace App\Profiler;
-
-    use Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface;
-    use Symfony\Component\HttpFoundation\Request;
-    use Symfony\Component\HttpFoundation\RequestMatcherInterface;
-
-    class SuperAdminMatcher implements RequestMatcherInterface
-    {
-        protected $authorizationChecker;
-
-        public function __construct(AuthorizationCheckerInterface $authorizationChecker)
-        {
-            $this->authorizationChecker = $authorizationChecker;
-        }
-
-        public function matches(Request $request)
-        {
-            return $this->authorizationChecker->isGranted('ROLE_SUPER_ADMIN');
-        }
-    }
-
-Then, you'll need to make sure your class is defined as as service. If you're using
-the :ref:`default services.yaml configuration <service-container-services-load-example>`,
-you don't need to do anything!
-
-Once the service is registered, the only thing left to do is configure the
-profiler to use this service as the matcher:
-
-.. configuration-block::
-
-    .. code-block:: yaml
-
-        # app/config/config.yml
-        framework:
-            # ...
-            profiler:
-                matcher:
-                    service: App\Profiler\SuperAdminMatcher
-
-    .. code-block:: xml
-
-        <!-- app/config/config.xml -->
-        <?xml version="1.0" encoding="UTF-8" ?>
-        <container xmlns="http://symfony.com/schema/dic/services"
-            xmlns:framework="http://symfony.com/schema/dic/symfony"
-            xmlns:xsi="http://www.w3.org/2001/XMLSchema-Instance"
-            xsi:schemaLocation="http://symfony.com/schema/dic/services
-                http://symfony.com/schema/dic/services/services-1.0.xsd
-                http://symfony.com/schema/dic/symfony
-                http://symfony.com/schema/dic/symfony/symfony-1.0.xsd">
-
-            <framework:config>
-                <!-- ... -->
-                <framework:profiler>
-                    <framework:matcher service="App\Profiler\SuperAdminMatcher" />
-                </framework:profiler>
-            </framework:config>
-        </container>
-
-    .. code-block:: php
-
-        // app/config/config.php
-        use App\Profiler\SuperAdminMatcher;
-
-        $container->loadFromExtension('framework', array(
-            // ...
-            'profiler' => array(
-                'matcher' => array(
-                    'service' => SuperAdminMatcher::class,
-                )
-            ),
-        ));
+    The possibility to use a matcher to enable the profiler conditionally was
+    removed in Symfony 4.0.
```

Comment:
We should add information about doing `$profiler->disable()` in your controller explicitly if you need this functionality.